oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/KeycloakRBACAuthorizer.java

```diff
@@ -0,0 +1,416 @@
+ * </li>
+ * </ul>
+ * <p>
+ * TLS configuration:
+ * </p>
+ * <ul>
+ * <li><em>strimzi.authz.ssl.truststore.location</em> The location of the truststore file on the filesystem.<br>
+ * If not present, <em>oauth.ssl.truststore.location</em> is used as a fallback configuration key to avoid unnecessary duplication when already present for the purpose of client authentication.
+ * </li>
+ * <li><em>strimzi.authz.ssl.truststore.password</em> The password for the truststore.<br>
+ * If not present, <em>oauth.ssl.truststore.password</em> is used as a fallback configuration key to avoid unnecessary duplication when already present for the purpose of client authentication.
+ * </li>
+ * <li><em>strimzi.authz.ssl.truststore.type</em> The truststore type.<br>
+ * If not present, <em>oauth.ssl.truststore.type</em> is used as a fallback configuration key to avoid unnecessary duplication when already present for the purpose of client authentication.
+ * If not set, the <a href="https://docs.oracle.com/javase/8/docs/api/java/security/KeyStore.html#getDefaultType--">Java KeyStore default type</a> is used.
+ * </li>
+ * <li><em>strimzi.authz.ssl.secure.random.implementation</em> The random number generator implementation. See <a href="https://docs.oracle.com/javase/8/docs/api/java/security/SecureRandom.html#getInstance-java.lang.String-">Java SDK documentation</a>.<br>
+ * If not present, <em>oauth.ssl.secure.random.implementation</em> is used as a fallback configuration key to avoid unnecessary duplication when already present for the purpose of client authentication.
+ * If not set, the Java platform SDK default is used.
+ * </li>
+ * <li><em>strimzi.authz.ssl.endpoint.identification.algorithm</em> Specify how to perform hostname verification. If set to empty string the hostname verification is turned off.<br>
+ * If not present, <em>oauth.ssl.endpoint.identification.algorithm</em> is used as a fallback configuration key to avoid unnecessary duplication when already present for the purpose of client authentication.
+ * If not set, the default value is <em>HTTPS</em> which enforces hostname verification for server certificates.
+ * </li>
+ * </ul>
+ * <p>
+ * This authorizer honors the <em>super.users</em> configuration. Super users are automatically granted any authorization request.
+ * </p>
+ */
+public class KeycloakRBACAuthorizer extends SimpleAclAuthorizer {
+
+    static final Logger log = LoggerFactory.getLogger(KeycloakRBACAuthorizer.class);
+
+    private URI tokenEndpointUrl;
+    private String clientId;
+    private String clusterName;
+    private SSLSocketFactory socketFactory;
+    private HostnameVerifier hostnameVerifier;
+    private List<UserSpec> superUsers = Collections.emptyList();
+    private boolean delegateToKafkaACL = false;
+
+
+    public KeycloakRBACAuthorizer() {
+        super();
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        super.configure(configs);
+
```

Comment:
Since using this authorizer requires using the principal builder I suppose you could check it here and throw if the principal builder is wrong.

Reply:
The check has to be performed in authorize() call in any case. But we can add it here as well.